From 2ca698e0e153055ce753bdc8d8027847a2133d71 Mon Sep 17 00:00:00 2001
From: tatataeki <shengzeyu19_98@163.com>
Date: Tue, 14 Sep 2021 17:57:37 +0800
Subject: [PATCH 1/5] add some basic statement pull in some header file
 essential for the communication and add some definition for the netlink
 protocol

Signed-off-by: tatataeki <shengzeyu19_98@163.com>
---
 kernel/signal.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/kernel/signal.c b/kernel/signal.c
index ef8f2a28d37c..63783912a98c 100644
--- a/kernel/signal.c
+++ b/kernel/signal.c
@@ -56,6 +56,10 @@
 #include <asm/siginfo.h>
 #include <asm/cacheflush.h>
 
+#include <net/sock.h>
+#include <net/netlink.h>
+#include <linux/ctype.h>
+
 /*
  * SLAB caches for signal bits.
  */
@@ -4637,3 +4641,17 @@ void kdb_send_sig(struct task_struct *t, int sig)
 		kdb_printf("Signal %d is sent to process %d.\n", sig, t->pid);
 }
 #endif	/* CONFIG_KGDB_KDB */
+
+/*
+ * signal monitor fragment
+ */
+
+#define NETLINK_SIG_MONITOR 25
+
+#define SIG_RECV_PID 100
+
+#define MAX_MSGSIZE 100
+
+struct sock *sig_monitor_sk = NULL;
+
+extern struct net init_net;
-- 
2.27.0

